src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1067,16 +1125,20 @@ UCS_CLASS_CLEANUP_FUNC(uct_dc_mlx5_ep_t)
      * Need to change if eps share dci
      */
     ucs_arbiter_group_cleanup(uct_dc_mlx5_ep_arb_group(iface, self));
-    ucs_assertv_always(uct_dc_mlx5_iface_dci_has_outstanding(iface, self->dci),
-                       "iface (%p) ep (%p) dci leak detected: dci=%d", iface,
-                       self, self->dci);
+
+    uct_dc_mlx5_ep_check_dci_leak(self);
 
     /* TODO should be removed by flush */
     uct_rc_txqp_purge_outstanding(&iface->super.super,
                                   &iface->tx.dcis[self->dci].txqp, UCS_ERR_CANCELED,
                                   iface->tx.dcis[self->dci].txwq.sw_pi, 1);
     ucs_assert(ucs_queue_is_empty(&iface->tx.dcis[self->dci].txqp.outstanding));
-    iface->tx.dcis[self->dci].ep = NULL;
+
+    if (!uct_dc_mlx5_iface_dci_has_outstanding(iface, self->dci)) {
```

Comment:
when is it possible? If ep is on a waitqueue? But then we invoke pending dispatch immediately, so there should be something outstanding